### PR TITLE
Feature/add better sqilte3

### PR DESCRIPTION
**Pull request by yennsarah**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -1,9 +1,88 @@
...
+// language=sql
+let createValidationLogsTableQuery = db.prepare(`
+    CREATE TABLE IF NOT EXISTS validation_logs(
+        id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT UNIQUE,
+        timestamp number,
+        valid bit (1),
+        body string,
+        schema string)`)
+
+let createErrorsTableQuery = db.prepare(`
```
Comment (dobschal) [inline routes/index.js:27]: @yennsarah please use ```const``` always, just if type of object changes or it's a primitive value that changes, use ```let```.

File: routes/index.js [hunk 2]
```diff
@@ -1,9 +1,88 @@
...
+}
+
+// language=sql
+let createValidationLogsTableQuery = db.prepare(`
+    CREATE TABLE IF NOT EXISTS validation_logs(
+        id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT UNIQUE,
+        timestamp number,
+        valid bit (1),
+        body string,
+        schema string)`)
```
Comment (dobschal) [inline routes/index.js:25]: @yennsarah for db migration things: lets have a separate folder, with a script file per migration step like "create_table_for_bla.js". Then an extra sql table "db_migration" is storing the migration scripts that already ran.

So on app startup you just execute the new added unknown db migration scripts. :)

File: routes/index.js [hunk 3]
```diff
@@ -1,9 +1,88 @@
...
+});
+
+router.post('/test-tv4', function (req, res, next) {
+    var body = req.body
+    var result = tv4.validateMultiple(body, schema)
+    saveToDb(body, schema, result)
+    res.send(result)
 });
 
+function saveToDb(body, schema, result) {
```
Comment (dobschal) [inline routes/index.js:53]: @yennsarah damn thats yummy!

File: routes/index.js [hunk 4]
```diff
@@ -1,9 +1,88 @@
...
+    // language=sql
+    let insertTestResultQuery = db.prepare('INSERT INTO validation_logs (timestamp, valid, body, schema) VALUES (@timestamp, @valid, @body, @schema)')
+    let insertTestResultQueryResult = insertTestResultQuery.run({
+        timestamp: Date.now(),
+        valid: result.valid ? 1 : 0,
+        body: JSON.stringify(body),
+        schema: JSON.stringify(schema)
+    })
+    console.log(insertTestResultQueryResult)
```
Comment (dobschal) [inline routes/index.js:62]: @yennsarah I always put the file at the beginning of the console.log, so we know later where its coming from. Like:
```javascript
console.log(`[routes/index] Got test result for '${superVariable}': `, testResult);
```

**Review by dobschal — CHANGES_REQUESTED** [review]

File: routes/index.js (hunk 2 above)
Comment (yennsarah) [reply routes/index.js:25]: About the migrations table:
What do I need to store there - just the name of the migration file? I haven't done anything like that before :)

On app startup, looping through the files in the folder and checking if one isn't int he database yet and then execute it?

**Pull request closed without merging**